src/effects/effectslot.cpp

```diff
@@ -146,9 +147,7 @@ void EffectSlot::loadEffect(EffectPointer pEffect) {
         m_pControlNumParameters->forceSet(pEffect->numKnobParameters());
         m_pControlNumButtonParameters->forceSet(pEffect->numButtonParameters());
 
-        // Enabled is a persistent property of the effect slot, not of the
-        // effect. Propagate the current setting to the effect.
-        pEffect->setEnabled(m_pControlEnabled->get() > 0.0);
+        pEffect->updateEngineState();
```

Comment:
If the effect slot is enabled then the effect will be disabled but the effect slot will be enabled.

It should be easy to see -- enable an effect, then change the effect to something else. The enabled button will be lit but the effect will be disabled.